packages/moonstone/Button/Button.js

```diff
@@ -86,7 +86,7 @@ const ButtonBase = kind({
 
 	styles: {
 		css: componentCss,
-		publicClassNames: ['button', 'bg', 'selected', 'small']
+		publicClassNames: ['button', 'bg', 'selected', 'small', 'transparent']
```

Comment:
Should we add all the bg opacity options to this list? Between this and the color change, we're starting to "onesie, twosie" class names onto `publicClassNames` and want to be sure we're making intentional choices. Either way, we'll need to update the `css` docs as well.

Reply:
`transparent` was intentionally excluded from the docs. The reasoning being that an app developer shouldn't need to dynamically alter the rules of the IconButton they're using. They can simply assign props and CSS rules as desired. For example, if an app wanted to alter the rules of their transparent button, they'd simply assign `backgroundOpacity="transparent"` and then add rules to their Button's `className`. If they wanted to change to non-transparent, they'd just remove that prop. It's not technically a useful feature for an app.

The framework, on the other hand, needs to be able to conditionally change styling rules without altering props. This is the most direct way to have IconButton respond to the presence of the `transparent` class. At this time we do not offer a `privateClassNames` option.

Reply:
Agree that it doesn't add value but I think it reveals a potential architectural problem when we're using features intended to expose public API but intentionally hiding them. It's not the first time we've done similar things (e.g. private props on base components) but I want to think a bit more on the implications of it.

Reply:
Fair point. Thanks for bringing it up. Much better to discuss now than when it's too late.

Reply:
Leaving this as is for now.